singularity-eos/eos/eos_eospac.hpp

```diff
@@ -340,9 +362,35 @@ class EOSPAC : public EosBase<EOSPAC> {
     EOS_REAL *dPdT = scratch + 1 * num;
 
     EOS_INTEGER table = PofRT_table_;
-    EOS_INTEGER options[]{EOS_F_CONVERT, EOS_XY_PASSTHRU};
-    EOS_REAL values[]{pressureFromSesame(1.0), 1.0};
-    EOS_INTEGER nopts = 2;
+    EOS_INTEGER options[3];
+    EOS_REAL values[3];
+    EOS_INTEGER nopts = 0;
+
+    if (!transform.x.is_set() && !transform.y.is_set()) {
+      options[nopts] = EOS_XY_PASSTHRU;
+      values[nopts] = 1.0;
+      ++nopts;
+    } else {
+      if (transform.x.is_set()) {
+        options[nopts] = EOS_X_CONVERT;
+        values[nopts] = 1.0 / transform.x.get();
+        ++nopts;
+      }
+
+      if (transform.y.is_set()) {
+        options[nopts] = EOS_Y_CONVERT;
+        values[nopts] = 1.0 / transform.y.get();
+        ++nopts;
+      }
+    }
```

Comment:
This is clever---let EOSPAC do the work by just setting options, rather than us doing the loops. I approve.